Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -556,6 +605,19 @@ TEST_F(SerializerTest, BadNullValue) {
       Status(FirestoreErrorCode::DataLoss, "ignored"), bytes);
 }
 
+/* TODO(rsgowman): If the number (representing a bool) on the wire is not 1 or
+ * 0, then nanopb happily drops the value into the bool anyways, but does so via
+ * an integer pointer. Something like this:
+ *   bool b;
+ *   *(int*)b = 2;
+ * Depending on the compiler, the resulting bool could be *both* true *and*
+ * false. eg at least gcc does this. (Effectively, comparing to true is
+ * implemented via != 0, and comparing to false is implemented via != 1.)
+ *
+ * We need to figure out what, if anything, we want to do about this. For now,
+ * we'll just ignore this test.
+ */
+#if 0
```

Comment:
Checking in `#if 0`ed code seems non-useful.

Can we make the Serializer do something like `static_cast<int>(proto.bool_value) != 0`? While not strictly to spec this would avoid this particular problem and remove something like 25 lines of comments :-).

Reply:
On some level whatever nanopb can do to fix this we can do too. If we can't fix it I'm not sure they could either.

We can avoid this crappy compiler behavior by memcpying from the source bool to a target int. A convenient implementation of this is in `absl::bit_cast` from [casts.h](https://github.com/abseil/abseil-cpp/blob/master/absl/base/casts.h).

In any case, commented out code is non-useful because it tends to rot. If we can't actually fix this then we should check the test in marked disabled, i.e. as `TEST_F(SerializerTest, DISABLED_BadBoolValue)`.

Reply:
bit_cast works nicely; done.